room-battle.js

```diff
@@ -886,7 +886,11 @@ class Battle {
 		if (this.started) this.onUpdateConnection(user);
 		if (this.p1 && this.p2) {
 			this.started = true;
-			Rooms.global.onCreateBattleRoom(Users(this.p1.userid), Users(this.p2.userid), this.room, {rated: this.rated});
+			const user1 = Users(this.p1.userid);
```

Comment:
`Users(...)` intentionally asserts the user exists so this sort of thing doesn't need to happen.

Reply:
Like, `Users(...)` claims to return `User` instead of `User?`

Reply:
That doesn't appear to be the case https://github.com/Zarel/Pokemon-Showdown/blob/master/users.js#L118.

Reply:
Hm, nevermind then.

I miss Real TypeScript where I can just do `!` to mean "stop complaining about nullables"...

Reply:
Petition to just actually write PS in TypeScript

Reply:
How's the error message situation? Chrome DevTools can apparently handle source maps, but I don't know if Node's built-in error reporting can.

Reply:
``Error#stack`` returns the code as-is. V8 knows nothing about source maps.

Reply:
Well, Babel 7 has a TypeScript transform _and_ the ability to preserve line numbers... I wonder...